### DISCUSSION TITLE
graph-learn功能空缺点

### DISCUSSION THREAD

**Discussion by bookug** [alibaba/GraphScope#1383] [Q&A] [answered]
请问现在graph-learn是否支持spark分布式训练？多机多GPU分布式训练？
是否支持TransformerConv，KGCN等较新的算法？
egograph是否可以对接pyg，或者像subgraph那样通过GSL转换数据集从而支持pyg?
训练/预测时各个点是独立训练的么，除了需要共享参数？
每个点维护自身的k跳邻居信息，这种做法是否会导致较大的内存开销？（是否有量化的统计数据）
PS: github上有独立的graph-learn仓库，这两边有区别么？会实时同步？

**Comment by sighingnow** [comment, marked as answer]
> 请问现在graph-learn是否支持spark分布式训练？

graph-learn provides efficient support for sampling (`graphscope.learning.Graph()`), and tensorflow is the builtin supported ML framework, and definitely other engines can be leveraged on top of graphlearn.

> 多机多GPU分布式训练？ 

Yes, distributed training is supported and multi-gpu the train depends on the framework ifself. It would require certain setup effort, IMO.

> 是否支持TransformerConv，KGCN等较新的算法？ 

GraphScope (as well as graphlearn) doesn't have those builtin algorihtms, but users definitely can implement such ones on top of current framework.

> egograph是否可以对接pyg，或者像subgraph那样通过GSL转换数据集从而支持pyg? 

graphlearn hasn't been integrated with pyg yet.

> 训练/预测时各个点是独立训练的么，除了需要共享参数？ 

Yes.

> 每个点维护自身的k跳邻居信息，这种做法是否会导致较大的内存开销？（是否有量化的统计数据） 

We haven't profiling yet and cannot tell such numbers currently.

> PS: github上有独立的graph-learn仓库，这两边有区别么？会实时同步？

GraphScope requires the [`graphscope` branch](https://github.com/alibaba/graph-learn/tree/graphscope) of the graph-learn repo, and depends it as a submodule.
